config/webhook.yaml

```diff
@@ -18,44 +18,78 @@ metadata:
   # Note: the Deployment name must be the same as the Service name specified in
   # config/400-webhook-service.yaml. If you change this name, you must also
   # change the value of WEBHOOK_SERVICE_NAME below.
-  name: tekton-pipelines-webhook
+  name: webhook
   namespace: tekton-pipelines
   labels:
     app.kubernetes.io/name: tekton-pipelines
     app.kubernetes.io/component: webhook-controller
-    tekton.dev/release: "devel"
+    pipeline.tekton.dev/release: "devel"
 spec:
   replicas: 1
   selector:
     matchLabels:
-      app: tekton-pipelines-webhook
+      app: webhook
+      role: webhook
   template:
     metadata:
       annotations:
         cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
-        tekton.dev/release: "devel"
       labels:
-        app: tekton-pipelines-webhook
+        app: webhook
+        role: webhook
         app.kubernetes.io/name: tekton-pipelines
         app.kubernetes.io/component: webhook-controller
+        pipeline.tekton.dev/release: "devel"
     spec:
       serviceAccountName: tekton-pipelines-controller
       containers:
       - name: webhook
         # This is the Go import path for the binary that is containerized
         # and substituted here.
         image: github.com/tektoncd/pipeline/cmd/webhook
-        volumeMounts:
-        - name: config-logging
-          mountPath: /etc/config-logging
         env:
         - name: SYSTEM_NAMESPACE
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        - name: CONFIG_LOGGING_NAME
+          value: config-logging
+        - name: CONFIG_OBSERVABILITY_NAME
+          value: config-observability
         - name: WEBHOOK_SERVICE_NAME
-          value: tekton-pipelines-webhook
-      volumes:
-        - name: config-logging
-          configMap:
-            name: config-logging
+          value: webhook
+        - name: METRICS_DOMAIN
+          value: tekton.dev/pipeline
+        securityContext:
+          allowPrivilegeEscalation: false
```

Comment:
You should change `.ko.yaml` to match the defaultBaseImage from serving, which uses `:nonroot`